addons/stock/migrations/13.0.1.1/post-migration.py

```diff
@@ -344,24 +355,35 @@ def map_stock_locations(env):
             imd.model = 'stock.location' and imd.res_id = sl.id)
         LEFT JOIN res_users ru ON sl.create_uid = ru.id
         WHERE tab.{location} = {value}sl.id AND
+            tab.company_id != {main_company_id} AND
             imd.name = '{xmlid_name}'
-                    """.format(table=table,
+                    """.format(table=table, main_company_id=main_company[0],
                                location=location, value=value,
                                xmlid_name=xmlid_name, condition=condition)
                 )
 
+    # avoid to remove records in end-migration scripts
+    # and some xmlids are deprecated in v13
+    openupgrade.logged_query(env.cr, """
+    DELETE FROM ir_model_data imd
+    WHERE imd.module = 'stock' AND imd.name IN (
+        'property_stock_inventory', 'property_stock_production',
+        'stock_location_scrapped', 'location_inventory', 'location_production')
+    """)
+
 
 @openupgrade.migrate()
 def migrate(env, version):
+    main_company = _get_main_company(env.cr)
     fill_company_id(env.cr)
     fill_stock_putaway_rule_location_in_id(env)
     fill_propagate_date_minimum_delta(env)
     fill_stock_inventory_start_empty(env)
     fill_stock_inventory_fields(env)
     map_stock_location_usage(env)
     fill_stock_picking_type_sequence_code(env)
-    handle_stock_scrap_sequence(env)
-    map_stock_locations(env)
+    handle_stock_scrap_sequence(env, main_company)
+    map_stock_locations(env, main_company)
     convert_many2one_stock_inventory_product_and_location(env)
     openupgrade.delete_records_safely_by_xml_id(env, _unlink_by_xmlid)
```

Comment:
delete this and also delete the end-migration

Reply:
Done